### PR TITLE
docker: Restore Firestore emulator

### PR DESCRIPTION
**Pull request by webvictim**
Thanks to [an oversight](https://github.com/gravitational/teleport/pull/6330/files#diff-1e5924d92807f9242cbbcdd715eb80a2237f28d8f4642a1d987df5b9c04a0811) in #6330, we haven't had the Firestore emulator installed and working inside the Docker buildbox on `master` between 8th April and now. As the tests just fail gracefully and don't run if the emulator isn't started, CI didn't fail and so nobody realised. I just happened to notice a `gcloud: command not found` while staring at unrelated build logs.

This restores the Firestore emulator to the buildbox and makes a failure to start `gcloud` inside the buildbox a hard failure.

### CODE REVIEW

File: build.assets/Makefile [hunk 1]
```diff
@@ -201,7 +201,7 @@ test: buildbox
 		$(DOCKERFLAGS) $(BCCFLAGS) $(NOROOT) -t $(BUILDBOX) \
 		/bin/bash -c \
 		"examples/etcd/start-etcd.sh & sleep 1; \
-		gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1; \
```
Comment (wadells) [inline build.assets/Makefile:204]: Per chesterson's fence, is it safe to remove the sleep?  This seems unrelated to the other changes.

**Review by wadells — APPROVED** [review]

File: build.assets/Makefile (hunk 1 above)
Comment (webvictim) [reply build.assets/Makefile:204]: You make a good point. I thought I'd remove it as it wasn't doing anything, but doing so meant that the emulator doesn't start in the background and so the tests don't actually run! Restored in da52cbd.

File: build.assets/Makefile [hunk 2]
```diff
@@ -201,7 +201,7 @@ test: buildbox
...
+		gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1 || exit 1; \
```
Comment (wadells) [inline build.assets/Makefile:204]: This || will chain off the sleep, rendering it meaningless:

```console
$ gcloud beta emulators firestore start --host-port=localhost:8618 & sleep 1 || echo hey
[1] 379596
ERROR: (gcloud.beta.emulators.firestore.start) To use the Google Cloud Firestore emulator, a Java 8+ JRE must be installed and on your system PATH
[1]  + exit 1     gcloud beta emulators firestore start --host-port=localhost:8618
```

I'd suggest a query, then a start in the background:

```console
$ gcloud beta emulators firestore start --help > /dev/null
$ cloud beta emulators firestore start --host-port=localhost:8618 &
$ sleep 1
...
```

**Review by wadells — CHANGES_REQUESTED** [review]

File: build.assets/Makefile (hunk 2 above)
Comment (webvictim) [reply build.assets/Makefile:204]: How about this?

```bash
type gcloud 2>&1 >/dev/null || exit 1; \
gcloud -q beta emulators firestore start --host-port=localhost:8618 & sleep 1; \
```

It'll error out if `gcloud` isn't installed, and the `-q` means that if the `beta` command set isn't installed for gcloud, it'll also install it (this shouldn't ever happen, but just in case). I can't find a way to make `gcloud` exit if it's not installed rather than auto-installing.

Comment (wadells) [reply build.assets/Makefile:204]: LGTM.

**Review by wadells — APPROVED** [review]

**Review by awly — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Comment by webvictim** [comment]
This PR started failing tests consistently after https://github.com/gravitational/teleport/pull/6731 was merged:

```

FAIL: firestoreevents_test.go:105: FirestoreeventsSuite.TestSessionEventsCRUD
--
450 |
451 | firestoreevents_test.go:106:
452 | s.SessionEventsCRUD(c)
453 | /go/src/github.com/gravitational/teleport/lib/events/test/suite.go:177:
454 | c.Assert(history, check.HasLen, 2)
455 | ... obtained []events.AuditEvent = []events.AuditEvent{(*events.UserLogin)(0xc0007542c0)}
456 | ... n int = 2
457 |
458 | OOPS: 0 passed, 1 FAILED
459 | --- FAIL: TestFirestoreevents (1.16s)


```

**Comment by webvictim** [comment]
Needs backports to `branch/v6.2` and `branch/v6`.
